src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
-                        typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        nameof(IConfigurationBuilder),
+                        nameof(IFileProvider)));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
+            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(fileProvider.GetFileInfo("/").PhysicalPath));
```

Comment:
Note that `AddUserSecrets()` already configures the underlying Json configuration provider to consider the user secrets file as optional around line 120. 

Anyway, @DamianEdwards seemed to feel no-op was the right behavior so I would ping him.